Connected/Assets/Scripts/Components/Battery.cs

```diff
@@ -3,9 +3,15 @@
 using UnityEngine;
 
 public class Battery : PowerSource {
+    [SerializeField]
+    float Voltage = 12f;
 
-    public Battery(float volt, float ampere) {
-        voltage = volt;
-        current = ampere;
+    [SerializeField]
+    float Ampere = 1f;
+
+    void Awake()
+    {
+        voltage = Voltage;
+        current = Ampere;
```

Comment:
The current is supposed to be set when the circuit is being traced and resolved, there is no need to set it in Awake, or have it be controlled through the editor in a serialized field.

Reply:
Makes sense.